packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/bulk_delete.test.ts

```diff
@@ -0,0 +1,603 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/* eslint-disable @typescript-eslint/no-shadow */
+
+import {
+  pointInTimeFinderMock,
+  mockGetBulkOperationError,
+  mockGetCurrentTime,
+  mockDeleteLegacyUrlAliases,
+  mockGetSearchDsl,
+} from '../repository.test.mock';
+
+import type { Payload } from '@hapi/boom';
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+import type {
+  SavedObjectsBulkDeleteObject,
+  SavedObjectsBulkDeleteOptions,
+} from '@kbn/core-saved-objects-api-server';
+import { ALL_NAMESPACES_STRING } from '@kbn/core-saved-objects-utils-server';
+import { SavedObjectsErrorHelpers } from '@kbn/core-saved-objects-server';
+import { SavedObjectsRepository } from '../repository';
+import { loggerMock } from '@kbn/logging-mocks';
+import { SavedObjectsSerializer } from '@kbn/core-saved-objects-base-server-internal';
+import { kibanaMigratorMock } from '../../mocks';
+import { elasticsearchClientMock } from '@kbn/core-elasticsearch-client-server-mocks';
+
+import {
+  NAMESPACE_AGNOSTIC_TYPE,
+  MULTI_NAMESPACE_TYPE,
+  MULTI_NAMESPACE_ISOLATED_TYPE,
+  HIDDEN_TYPE,
+  mockTimestamp,
+  mappings,
+  createRegistry,
+  createDocumentMigrator,
+  getMockMgetResponse,
+  type TypeIdTuple,
+  createSpySerializer,
+  expectErrorInvalidType,
+  expectErrorNotFound,
+  createBadRequestErrorPayload,
+  getMockEsBulkDeleteResponse,
+  bulkDeleteSuccess,
+  createBulkDeleteSuccessStatus,
+} from '../../test_helpers/repository.test.common';
+
+// BEWARE: The SavedObjectClient depends on the implementation details of the SavedObjectsRepository
+// so any breaking changes to this repository are considered breaking changes to the SavedObjectsClient.
```

Comment:
NIT: no really sure why that comment was on the SOR test file in the first place, seems like something that was very old. I think we should remove it from all those files now.

Reply:
Done: https://github.com/elastic/kibana/pull/173218/commits/8276f13283f644c13ce38b7e53ce769edd67686f